pkg/controller/cloud/nodecontroller.go

```diff
@@ -55,6 +65,13 @@ const (
 
 	// The amount of time the nodecontroller should sleep between retrying NodeStatus updates
 	retrySleepTime = 20 * time.Millisecond
+
+	//Taint denoting that a node needs to be processed by external cloudprovider
+	CloudTaintKey = "ExternalCloudProvider"
+
+	nodeStatusUpdateFrequency = 10 * time.Second
+
+	LabelProvidedIPAddr = "beta.kubernetes.io/provided-node-ip"
```

Comment:
I would vote for an annotation for this.

But if we don't understand the use-case for this, I would maybe say that it's best to error out if both nodeIP and cloudprovider=external is defined and see if it causes trouble to people. To be honest I'm having a hard time seeing that someone would want to set a specific node IP on every kubelet's cmdline and still verify/lookup things in the cloud provider DB.

If we would restrict the nodeIP && external cloudprovider possibility now, we would end up adding less code and that would be great. The less we add in this refactoring phase the better.

However cc-ing @kubernetes/sig-node-pr-reviews